spec/02-integration/05-proxy/03-plugins_triggering_spec.lua

```diff
@@ -1000,5 +1001,111 @@ for _, strategy in helpers.each_strategy() do
         assert.equal("timeout", res.headers["Log-Plugin-Service-Matched"])
       end)
     end)
+
+    describe("plugin with run_on", function()
+      lazy_setup(function()
+        if proxy_client then
+          proxy_client:close()
+        end
+
+        helpers.stop_kong()
+
+        db:truncate("routes")
+        db:truncate("services")
+        db:truncate("plugins")
+
+        do
+          local service = assert(bp.services:insert {
+            name = "example",
+            host = helpers.mock_upstream_host,
+            port = helpers.mock_upstream_port,
+          })
+
+          local route1 = assert(db.routes:insert {
+            hosts     = { "run-on-second.org" },
+            protocols = { "http" },
+            service   = service,
+          })
+
+          assert(bp.plugins:insert {
+            name = "error-generator-pre",
+            route = { id = route1.id },
+            config = {
+              access = true,
+            },
+            run_on = "second",
+          })
+
+          local route2 = assert(db.routes:insert {
+            hosts     = { "run-on-first.org" },
+            protocols = { "http" },
+            service   = service,
+          })
+
+          assert(bp.plugins:insert {
+            name = "error-generator-pre",
+            route = { id = route2.id },
+            config = {
+              access = true,
+            },
+            run_on = "first",
+          })
+
+          local route3 = assert(db.routes:insert {
```

Comment:
It seems this route is not used on any tests.

Reply:
@kikito, yes, I forgot to push latest, it should be now there.